shared/studio/tabs/auth/state/index.tsx

```diff
@@ -368,6 +412,169 @@ export class DraftUIConfig extends Model({
   }
 }
 
+@model("AdminDraftSMTPConfig")
+export class DraftSMTPConfig extends Model({
+  _sender: prop<string | null>(null),
+  _host: prop<string | null>(null),
+  _port: prop<string | null>(null),
+  _username: prop<string | null>(null),
+  _password: prop<string | null>(null),
+  _security: prop<SMTPSecurity | null>(null),
+  _validate_certs: prop<boolean | null>(null),
+  _timeout_per_email: prop<string | null>(null),
+  _timeout_per_attempt: prop<string | null>(null),
+}) {
+  getConfigValue(name: Exclude<keyof SMTPConfigData, "validate_certs">) {
+    return (
+      this[`_${name}`] ??
+      (getParent<AuthAdminState>(this)?.smtpConfig || null)?.[name] ??
+      ""
+    );
+  }
+
+  @modelAction
+  setConfigValue<Name extends keyof SMTPConfigData>(
+    name: Name,
+    val: SMTPConfigData[Name]
+  ) {
+    (this as any)[`_${name}`] = val;
+  }
+
+  @computed
+  get portError() {
+    const val = this.getConfigValue("port").trim();
+    if (val === "") {
+      return null;
+    }
+    const int = parseInt(val, 10);
+    if (!/^\d+$/.test(val) || Number.isNaN(int) || int < 0 || int > 65535) {
+      return `Port must be an integer between 0 and 65535`;
+    }
+    return null;
+  }
+
+  @computed
+  get timeoutPerEmailError() {
+    const val = this.getConfigValue("timeout_per_email");
+    if (!val.length) {
+      return `Value is required`;
+    }
+    try {
+      parsers["std::duration"](val, null);
+    } catch {
+      return `Invalid duration`;
+    }
+    return null;
+  }
```

Comment:
Did we need to do anything here to get the same duration input we use for the `token_time_to_live` control where you specify the number of seconds?

Reply:
Yeah, I had forgot about that. I've updated it now.